flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1200,7 +1208,10 @@ fir::ExtendedValue IntrinsicLibrary::genElementalCall(
   }
   if (outline)
     return outlineInWrapper(generator, name, resultType, scalarArgs);
-  return invokeGenerator(generator, resultType, scalarArgs);
+  if constexpr (std::is_same_v<GeneratorType, SubroutineGenerator>)
+    return invokeGenerator(generator, scalarArgs);
+  else
+    return invokeGenerator(generator, resultType, scalarArgs);
```

Comment:
`invokeGenerator` on `SubroutineGenerator` works, but it is doing extra work because it is intended to handle outlining, and to start with `mlir::Value`. Outside of outlining, when starting from ExtendedValue, using `invokeGenerator` involves going from ExtendedValue to mlir::Value, and then back to ExtendedValue, which might generate extra boxing/unboxing for nothing (and is probably wrong her for characters since the Exv to mlir::Value should create a fir.boxchar). For MVBITS, it is probably transparent IR wise, but still adding some compile time work.

I would rather suggest making a new template specialization of IntrinsicLibrary::genElementalCall for `SubroutineGenerator` using ` std::invoke()` and using the ExtendedValue args directly like in `IntrinsicLibrary::genElementalCall<IntrinsicLibrary::ExtendedGenerator>`.

In general,  IntrinsicLibrary::genElementalCall was intended to handle array arguments in  the intrinsic lowering library, but this job was eventually given to the array lowering framework (the TODOs are a little bit dated, and should be replaced by fatal errors that only scalars should get in here today). Its only mission today is to verify that the arguments are indeed scalars, and to already extract mlir::Value from ExtendedValue for the `ElementalGenerator` case. Otherwise, it is pretty much a no-op.

It is kept in the flow in case we want at some point  to dispatch some elemental intrinsics to array runtime versions instead of generating the loop inlined and calling scalar runtime versions.

Reply:
I've added a new `genElementalCall` template specialization for `SubroutineGenerator` as suggested.  That bypasses the usual TODO message for an unimplemented intrinsic for an elemental MVBITS call.  There is a temporary `crashOnMissingIntrinsic` call to compensate for that.